Content.Server/Botany/Systems/MutationSystem.cs

```diff
@@ -0,0 +1,211 @@
+    [Dependency] private static readonly IRobustRandom _robustRandom = default!;
+
+    /// <summary>
+    // Main idea: Simulate genetic mutation using random binary flips.  Each
+    // seed attribute can be encoded with a variable number of bits, e.g.
+    // NutrientConsumption is represented by 5 bits randomly distributed in the
+    // plant's genome which thermometer code the floating value between 0.1 and
+    // 5. 1 unit of mutation flips one bit in the plant's genome, which changes
+    // NutrientConsumption if one of those 5 bits gets affected.
+    //
+    // You MUST clone() seed before mutating it!
+    /// </summary>
+    public static void MutateSeed(SeedData seed, float severity)
+    {
+        // Add up everything in the bits column and put the number here.
+        const int totalbits = 215;
+
+        // Tolerances (55)
+        MutateFloat(ref seed.NutrientConsumption   , 0.05f , 1.2f , 5 , totalbits , severity);
+        MutateFloat(ref seed.WaterConsumption      , 3f    , 9f   , 5 , totalbits , severity);
+        MutateFloat(ref seed.IdealHeat             , 263f  , 323f , 5 , totalbits , severity);
+        MutateFloat(ref seed.HeatTolerance         , 2f    , 25f  , 5 , totalbits , severity);
+        MutateFloat(ref seed.IdealLight            , 0f    , 14f  , 5 , totalbits , severity);
+        MutateFloat(ref seed.LightTolerance        , 1f    , 5f   , 5 , totalbits , severity);
+        MutateFloat(ref seed.ToxinsTolerance       , 1f    , 10f  , 5 , totalbits , severity);
+        MutateFloat(ref seed.LowPressureTolerance  , 60f   , 100f , 5 , totalbits , severity);
+        MutateFloat(ref seed.HighPressureTolerance , 100f  , 140f , 5 , totalbits , severity);
+        MutateFloat(ref seed.PestTolerance         , 0f    , 15f  , 5 , totalbits , severity);
+        MutateFloat(ref seed.WeedTolerance         , 0f    , 15f  , 5 , totalbits , severity);
+
+        // Stats (30*2 = 60)
+        MutateFloat(ref seed.Endurance             , 50f   , 150f , 5 , totalbits , 2*severity);
+        MutateInt(ref seed.Yield                   , 3     , 10   , 5 , totalbits , 2*severity);
+        MutateFloat(ref seed.Lifespan              , 10f   , 80f  , 5 , totalbits , 2*severity);
+        MutateFloat(ref seed.Maturation            , 3f    , 8f   , 5 , totalbits , 2*severity);
+        MutateFloat(ref seed.Production            , 1f    , 10f  , 5 , totalbits , 2*severity);
+        MutateFloat(ref seed.Potency               , 30f   , 100f , 5 , totalbits , 2*severity);
+
+        // Kill the plant (30)
+        MutateBool(ref seed.Viable         , false , 30 , totalbits , severity);
+
+        // Fun (70)
+        MutateBool(ref seed.Seedless       , true  , 10 , totalbits , severity);
+        MutateBool(ref seed.Slip           , true  , 10 , totalbits , severity);
+        MutateBool(ref seed.Sentient       , true  , 10 , totalbits , severity);
+        MutateBool(ref seed.Ligneous       , true  , 10 , totalbits , severity);
```

Comment:
any reason why `MutateBool` only goes for one side (i.e. why these can't be flipped off with a mutation?)

Reply:
To make it more likely that you get an observable boolean mutation. If there was one bit with an equal chance of flipping back and forth, it would tend towards a coin toss and be less interesting. With this bias, you are guaranteed to get more mutations as long as you add more unstable mutagen.

Seedless and Viable traits are intended to "nerf" mutation: seedless by making mutated plants not have seeds (you can't just clone them), and Viable to kill the plants outright. Having those particular traits mutate back would un-nerf the nerf.

Reply:
okay, that makes sense